tuna/connection.py

```diff
@@ -174,14 +257,18 @@ def exec_command_unparsed(self, cmd, timeout=SSH_TIMEOUT, abort=None):
       if abort is not None and chk_abort_file(self.id, self.logger):
         self.logger.warning('Machine %s aborted command execution: %s', self.id,
                             cmd)
-        return None, None, None
+        return i_var, o_var, e_var
 
     self.logger.error('cmd_exec retries exhausted, giving up')
-    return None, None, None
+    return i_var, o_var, e_var
 
-  def exec_command(self, cmd, timeout=SSH_TIMEOUT, abort=None, proc_line=None):
+  def exec_command(self, cmd: str, timeout: int=int(SSH_TIMEOUT), abort: Optional[bool]=None,\
+  proc_line: Union[Any, None]= None) -> Tuple[Union[int, Any], Union[TextIO, StringIO, \
+  ChannelFile],Union[ChannelStderrFile, StringIO]]:
```

Comment:
return type seems overcomplicated.
tuple val 1 : should always be int
tuple val 2 : appears to be exclusively StringIO
tuple val 3 : appears to be exclusively StringIO

Reply:
269 : o_var is defined as Channel_File , it is need in Tuple value 2

Reply:
connection.py:274: error: Incompatible return value type (got "Tuple[int, ChannelFile, ChannelStderrFile]", expected "Tuple[int, Union[ChannelFile, StringIO], StringIO]")
connection.py:308: error: Incompatible return value type (got "Tuple[Union[int, Any], StringIO, Union[ChannelStderrFile, StringIO]]", expected "Tuple[int, Union[ChannelFile, StringIO], StringIO]")
we need to handle these types. when we combine. retrun statement passes all the tests too. as exec parse return/makes assighment to different IO types, mypy does not support casting.

Reply:
Only in Tuple - 2 Text IO removed.